src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -328,21 +321,56 @@ private void SetStateForCustomizationTask(TaskJSONToCSClasses.BaseClass response
         }
     }
 
+    // Start the Dev Box if it isn't already running
+    private async Task HandleNonRunningState()
+    {
+        // Check if the dev box might have been started in the meantime
+        var stateResult = await _devBox.GetStateAsync();
+        if (stateResult.State == ComputeSystemState.Running)
+        {
+            return;
+        }
+
+        // Start the Dev Box
+        await _devBox.StartAsync(string.Empty);
+        _log.Information("Starting the dev box to apply configuration");
+
+        // Notify the user
+        ConfigurationSetStateChanged?.Invoke(this, new(new(ConfigurationSetChangeEventType.SetStateChanged, ConfigurationSetState.StartingDevice, ConfigurationUnitState.Unknown, null, null)));
+
+        // Wait for the Dev Box to start with an initial wait of 3 minutes and a max of 15 minutes
+        var delay = TimeSpan.FromSeconds(30);
+        var waitTimeLeft = TimeSpan.FromMinutes(12);
+        await Task.Delay(TimeSpan.FromMinutes(3));

```

Comment:
Do we need this 3-minute wait? Are chances that VM can start earlies is so low? The code below checks the state once in 30 seconds. I wouldn't think sending 1 request per 30 seconds is too often.

Reply:
The 3 minute delay is to accommodate the fact that a Dev Box won't start in 3 minutes and there is no use of polling the meanwhile. Let me know if you feel I should change it.

Reply:
My thinking is that DevBox can improve that in the future (or it could depend on the backend hardware, or customer's subscription), and we may not know about that until someone complains that Dev Home is waiting for too long after Dev Box was already started. And that will require another code change since this timeout is hardcoded.

Reply:
I see. That's a good future thinking point; changed.